sdk/servicebus/service-bus/src/core/managementClient.ts

```diff
@@ -301,29 +301,34 @@ export class ManagementClient extends LinkEntity<RequestResponseLink> {
     internalLogger: ServiceBusLogger,
     sendRequestOptions: SendManagementRequestOptions = {}
   ): Promise<RheaMessage> {
+    if (request.message_id == undefined) {
+      request.message_id = generate_uuid();
+    }
     const retryTimeoutInMs =
       sendRequestOptions.timeoutInMs ?? Constants.defaultOperationTimeoutInMs;
     const initOperationStartTime = Date.now();
-    const actionAfterTimeout = () => {
+    const actionAfterTimeout = (reject: (reason?: any) => void) => {
       const desc: string = `The request with message_id "${request.message_id}" timed out. Please try again later.`;
       const e: Error = {
         name: "OperationTimeoutError",
         message: desc
       };
 
-      throw e;
+      reject(e);
     };
 
-    const waitTimer = setTimeout(actionAfterTimeout, retryTimeoutInMs);
-
+    let waitTimer: NodeJS.Timer;
```

Comment:
I think it would be safer to set the type of `waitTimer` to something like this:
```suggestion
    let waitTimer: ReturnType<typeof setTimeout>;
```

This is because in browsers, you won't have the NodeJS.Timer type (it returns as a number instead). Though, since this isn't actually user-facing it won't matter to the end-user...so I leave it up to you whether to make this change.

Reply:
Hmmm.. updated